app/src/main/java/fr/free/nrw/commons/Media.java

```diff
@@ -64,14 +68,28 @@
      * Provides local constructor
      */
     public Media() {
+        pageId = UUID.randomUUID().toString();
     }
 
+    public static final Creator<Media> CREATOR = new Creator<Media>() {
+        @Override
+        public Media createFromParcel(final Parcel source) {
+            return new Media(source);
+        }
+
+        @Override
+        public Media[] newArray(final int size) {
+            return new Media[size];
+        }
+    };
+
     /**
      * Provides a minimal constructor
      *
      * @param filename Media filename
      */
-    public Media(String filename) {
+    public Media(final String filename) {
+        pageId = UUID.randomUUID().toString();
```

Comment:
you could replace this line with `this()`

Reply:
@macgills I didn't understand the requested change. Can you elaborate?

Reply:
`pageId = UUID.randomUUID().toString();` is repeated, you can invoke the parameter-less constructor by calling `this()` and stop the repetition.

Kotlin - because it is great - enforces this and all secondary constructors must delegate to the primary constructor.